### DISCUSSION TITLE
How to add new GraphQLObjectType Object into schema

### DISCUSSION THREAD

**Discussion by BlinglingGirl** [graphql-java/graphql-java#2634] [Q&A] [answered]
Hi,

I have a GraphQLSchema on which i need to add new GraphQLObjectType object. 
can I use graphql.util.TreeTransformerUtil#insertAfter API to insert the element? but, I don't care where elements are inserted

Many thanks

**Comment by bbakerman** [comment, marked as answer]
SchemaTransformer is what you want

```
        def fieldChanger = new GraphQLTypeVisitorStub() {

            @Override
            TraversalControl visitGraphQLObjectType(GraphQLObjectType node, TraverserContext<GraphQLSchemaElement> context) {
                if (node.name == "__Field") {
                    return changeNode(context, node.transform({ it.name("__FieldChanged") }))
                }
                return TraversalControl.CONTINUE;
            }
        }

        def newSchema = SchemaTransformer.transformSchema(schema, fieldChanger)
```

Adding just a new type is weird if no field references to it - did you mean you wanted to adda new field referencing a new type?

**Reply by BlinglingGirl** [reply]
I know. It doesn't make any sense. You are right, thanks!
